homeassistant/components/duotecno/entity.py

```diff
@@ -35,3 +40,26 @@ async def async_added_to_hass(self) -> None:
     async def _on_update(self) -> None:
         """When a unit has an update."""
         self.async_write_ha_state()
+
+
+_T = TypeVar("_T", bound="DuotecnoEntity")
+_P = ParamSpec("_P")
+
+
+def api_call(
+    func: Callable[Concatenate[_T, _P], Awaitable[None]]
+) -> Callable[Concatenate[_T, _P], Coroutine[Any, Any, None]]:
+    """Catch command exceptions."""
+
+    @wraps(func)
+    async def cmd_wrapper(self: _T, *args: _P.args, **kwargs: _P.kwargs) -> None:
+        """Wrap all command methods."""
+        try:
+            await func(self, *args, **kwargs)
+        except OSError as exc:
+            raise HomeAssistantError(
+                f"Error calling {func.__name__} on entity {self.entity_id},"
+                f" packet transmit failed"
```

Comment:
Can we make this slightly more user-friendly as it's shown in the UI? Perhaps just remove the last line as it doesn't mean anything to the user.